### PR TITLE
feat(nav-bar-content): move switcher to leftNav

### PR DESCRIPTION
**Pull request by pownkel**
#### Description of changes

Move the DetailsView switcher to the top of the LeftNav behind the feature flag and make necessary styling changes.

This is the old switcher (still available when feature flag is off):
switcher-old

the new switcher (unfocused and focused):
switcher
switcher-focused

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/left-nav/details-view-left-nav.tsx [hunk 1]
```diff
@@ -47,9 +52,19 @@ export const DetailsViewLeftNav = NamedFC<DetailsViewLeftNavProps>('DetailsViewL
...
+            deps: props.deps,
+            pivotKey: props.selectedPivot,
+            styles: leftNavSwitcherStyleNames,
+        };
+        switcher = <Switcher {...switcherProps} />;
+    }
 
     const leftNav: JSX.Element = (
         <div className="left-nav main-nav">
+            {switcher}
```
Comment (waabid) [inline src/DetailsView/components/left-nav/details-view-left-nav.tsx:67]: I believe you should be able to use the `FlaggedComponent` here?

File: src/DetailsView/components/switcher-style-names.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as leftNavSwitcherStyles from './left-nav-switcher.scss';
+import * as headerSwitcherStyles from './switcher.scss';
+
+export interface SwitcherStyleNames {
```
Comment (waabid) [inline src/DetailsView/components/switcher-style-names.ts:7]: This is a nice solution! 👍

File: src/DetailsView/details-view-body.tsx [hunk 3]
```diff
@@ -98,7 +98,11 @@ export class DetailsViewBody extends React.Component<DetailsViewBodyProps> {
             return null;
         }
 
-        return <DetailsViewLeftNav {...this.props} />;
+        const selectedPivot = this.props.visualizationStoreData
```
Comment (waabid) [inline src/DetailsView/details-view-body.tsx:101]: I think you should be able to replace this with:

`this.props.visualizationStoreData?.selectedDetailsViewPivot`

File: src/DetailsView/components/interactive-header.tsx [hunk 4]
```diff
@@ -22,7 +23,17 @@ export const InteractiveHeader = NamedFC<InteractiveHeaderProps>('InteractiveHea
...
+    const getItems = () => {
+        if (!props.featureFlagStoreData['reflowUI']) {
+            const switcherProps = {
+                deps: props.deps,
+                pivotKey: props.selectedPivot,
+                styles: headerSwitcherStyleNames,
+            };
+            return <Switcher {...switcherProps} />;
+        }
+        return null;
```
Comment (waabid) [inline src/DetailsView/components/interactive-header.tsx:35]: Similarly, you should be able to use flagged component here as well.

**Comment by waabid** [comment]
Based off of the screenshots you've posted, my understanding is that if the left nav is scrolled down, the switcher will disappear. I'm not sure if that's what we want: I would check with PM team on that and make the `switch` if necessary.

**Comment by waabid** [comment]
ooooh and the title should read `feat(nav-bar-content)`

**Comment by pownkel** [comment]
I confirmed with Fer that we do want the switcher to disappear when we scroll in the assessment view

**Pull request merged**